pkg/ruler/api_test.go

```diff
@@ -184,6 +363,14 @@ func TestRuler_ListRules(t *testing.T) {
 			} else {
 				require.Contains(t, string(body), tc.expectedErr)
 			}
+
+			// Check logs.
+			for _, expect := range tc.expectLogsContain {
+				require.Contains(t, logs.String(), expect)
+			}
+			for _, expect := range tc.expectLogsNotContain {
+				require.NotContains(t, logs.String(), expect)
```

Comment:
Can we come up with another way to test this? I really don't like introducing dependencies on particular log messages.

Reply:
I need suggestions :)

Reply:
I'm testing that we log it. I want to make sure we have a test :) I'm going to keep it. I see this kind of logs testing similar to metrics testing: just testing observability signals. Don't see it as a bad thing.